src/main/java/com/hubspot/jinjava/lib/tag/eager/EagerTagDecorator.java

```diff
@@ -38,29 +38,40 @@ public T getTag() {
   @Override
   public final String interpret(TagNode tagNode, JinjavaInterpreter interpreter) {
     try {
-      String output = innerInterpret(tagNode, interpreter);
-      JinjavaInterpreter.checkOutputSize(output);
-      return output;
-    } catch (DeferredValueException | TemplateSyntaxException | OutputTooBigException e) {
+      String output;
       try {
-        return EagerReconstructionUtils.wrapInAutoEscapeIfNeeded(
-          eagerInterpret(
-            tagNode,
-            interpreter,
-            e instanceof InterpretException
-              ? (InterpretException) e
-              : new InterpretException("Exception with default render", e)
-          ),
-          interpreter
-        );
-      } catch (OutputTooBigException e1) {
-        throw new DeferredValueException(
-          String.format("Output too big for eager execution: %s", e1.getMessage())
-        );
+        output = innerInterpret(tagNode, interpreter);
+      } catch (DeferredValueException | TemplateSyntaxException e) {
+        return wrapEagerInterpret(tagNode, interpreter, e);
       }
+      if (JinjavaInterpreter.isOutputTooLarge(output)) {
+        return wrapEagerInterpret(tagNode, interpreter, null);
+      }
```

Comment:
We only want to do the deferred execution eager interpret if we fully succeeded, but the output is too large. If we encountered an uncaught OutputTooBigException while running `innerInterpret`, we haven't handled that, and the context won't be in a proper state. (It likely would come from attempting to reconstruct a value, but its string representation is too large.)